resources/design-document.md

```diff
@@ -0,0 +1,212 @@
+
+### 6.3 Get One Chart Endpoint
+
+* Accepts `GET` request to `/chart/:id`
+* Accepts chart `ID` and Queries for specific chart and returns ChartModel
+* If chart not found, will throw `ChartNotFoundException`
+
+![getChart.png](Images%2Fapi_design_images%2FgetChart.png)
+
+### 6.4 Create Chart Endpoint
+
+* Accepts `POST` request to `/chart/:id` 
+* Will contain optional body to post other fields : "`name`, `artist`, `bpm`, `content`, `genres`"
+* Accepts chart `ID` and adds the chart to the `Charts` table
+  * `ID` will be generated using java's `UUID` class
+* If chart name/artist/Content contains invalid characters: `" ' \ `, InvalidAttributeException will be thrown
+* If chart is unable to be added to `Charts` table, will throw `UnableToAddToTableException`
+
+### 6.5 Update Chart Endpoint
+
+* Accepts `PUT` request to `/chart/:id`
+* Accepts chart `ID` and overrides old chart on `Charts` table
+* Will contain optional body to update fields : "`name`, `artist`, `bpm`, `content`, `genres`"
+* If chart name/artist/Content contains invalid characters: `" ' \ `, InvalidAttributeException will be thrown
+* If chart is unable to be added to `Charts` table, will throw `UnableToAddToTableException`
+
+![updateChart.png](Images%2Fapi_design_images%2FupdateChart.png)
+
+### 6.6 Create SetList Endpoint
+
+* Accepts `POST` request to `/setList/:id`
+* Will contain optional body to post other fields : "`name`, `genres`, `charts`"
+* Accepts setList `ID` and adds to `SetLists` table
+  * `ID` will be generated using java's `UUID` class
+* If setList name//Content contains invalid characters: `" ' \ `, InvalidAttributeException will be thrown
+* If setList is unable to be added to `SetLists` table, will throw `UnableToAddToTableException`
+
+![createSetList.png](Images%2Fapi_design_images%2FcreateSetList.png)
+
+### 6.7 Get one SetList  Endpoint
+
+* Accepts `GET` request to `/setlist/:id`
+* Accepts setList `ID` and Queries for specific setList and returns SetListModel
+* If setList not found, will throw `SetListNotFoundException`
+
+### 6.8 Add to SetList Endpoint
+
+* Accepts `POST` request to `/setList/:id/chart/:chartId`
+* Accepts setList `ID` and chart `ID` and adds chart to `SetLists` table
+* Will check to make sure chart exists in `Charts` table
+* If chart is unable to be added to `SetLists` table, will throw `UnableToAddToTableException`
+
+![addToSetList.png](Images%2Fapi_design_images%2FaddToSetList.png)
+
+### 6.9 Remove from SetList Endpoint
+
+* Accepts `PUT` request to `/setList/:id/chart/:chartId`
+* Accepts setList `ID` and chart `ID` and removes chart from `SetLists` table
+* Will update entry
+* Will check to make sure chart exists in `Charts` table
+* If chart is unable to be added to `SetLists` table, will throw `UnableToAddToTableException`
+
+// Unsure whether I need a model for PDF if I am storing them in an S3 bucket?
+### 6.10 Convert Chart to PDF Endpoint
+
+* Accepts `GET` request to `/pdf/chart/:id`
```

Comment:
Similar to an earlier note, I'd have this build on the URL path. So `/charts/:id/pdf` ... the URLs represent a hierarchy of sorts.